### PR TITLE
Tec member dashboard frontend

### PR DESCRIPTION
**Pull request by airaagrawal1**
### Summary 

This pull request displays team event credit information for DTI members to check their TEC status. 

### Test Plan 

Tested using mock data in dev firebase and all results were being displayed correctly. 

Screen Shot 2022-03-04 at 11 19 37 AM

### CODE REVIEW

**Review by henry-li-06 — COMMENTED** [review]
Would you be able to add screenshots? It's hard to reproduce what it looks like at the moment

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx [hunk 1]
```diff
@@ -63,6 +63,47 @@ const TeamEventCreditForm: React.FC = () => {
...
+          if (currMemberTECStatus.countRemainingCredits - Number(currTeamEvent.numCredits) < 0)
+            currMemberTECStatus.countRemainingCredits = 0;
+          else currMemberTECStatus.countRemainingCredits -= Number(currTeamEvent.numCredits);
+          if (currMemberTECStatus.approvedTEC.length === 0)
+            currMemberTECStatus.approvedTEC = currTeamEvent.name;
+          else
+            currMemberTECStatus.approvedTEC = `${currMemberTECStatus.approvedTEC}, ${currTeamEvent.name}`;
+        }
+      }
+      for (const currApprovedMember of currTeamEvent.requests) {
```
Comment (henry-li-06) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:94]: Use `forEach`

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx [hunk 2]
```diff
@@ -63,6 +63,47 @@ const TeamEventCreditForm: React.FC = () => {
...
+    approvedTEC: string;
+    pendingTEC: string;
+  };
+
+  const currMemberTECStatus: MemberTeamEventCreditStatus = {
+    countApprovedCredits: 0,
+    countRemainingCredits: 3,
+    approvedTEC: '',
+    pendingTEC: ''
+  };
```
[the remark this answers is not in the document]
Comment (airaagrawal1) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx]: Done!

**Comment by riyajaggi** [comment]
image
Since I don't have any approved/pending events, it shows an empty list. Instead, can we display some message like `None`/`No events`?

**Comment by airaagrawal1** [comment]
> image
> Since I don't have any approved/pending events, it shows an empty list. Instead, can we display some message like `None`/`No events`?

Good call! It is implemented!

**Comment by henry-li-06** [comment]
> empty

+1

**Comment by henry-li-06** [comment]
I think it might be better for the display of events to maybe be a card or something instead of just displaying at `[test1]`

**Review by henry-li-06 — APPROVED** [review]

**Pull request merged**